napari/layers/utils/style_encoding.py

```diff
@@ -0,0 +1,228 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+from pydantic import ValidationError, parse_obj_as, validator
+
+from ...utils import Colormap
+from ...utils.colormaps import ValidColormapArg, ensure_colormap
+from ...utils.colormaps.categorical_colormap import CategoricalColormap
+from ...utils.colormaps.standardize_color import transform_color
+from ...utils.events import EventedModel
+
+
+class StyleEncoding(EventedModel, ABC):
```

Comment:
I like the encoding name, you mentioned it comes from Altair - in general i've liked using that library a lot 😄